elytron/WFCORE-5744-ssl-certificate-expiration-warning.adoc

```diff
@@ -0,0 +1,218 @@
+
+
+=== User Stories
+
+Long runing server, with periodic information on certificates will allow admins to not only setup tools to warn them about potential security risk, but also schedule maintanence window based on this.
+
+== Issue Metadata
+
+
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1863[EAP7-1863]
+* https://issues.redhat.com/browse/WFCORE-5744[WFCORE-5744]
+
+=== Affected Projects or Components
+
+ * Elytron integration
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+ * create output format that is not driven by exception, so possibly it can be easily digested via log munching, mail notification or any other mechanism users see fit to employ over log handler.
+ * read-alias command will have additional KVP "Validity=[valid, EXPIRED, about to expire, not yet]"
+ ** Validity semantics:
+ *** not yet - ceritifacete is still too young
+ *** EXPIRED - current date is after expiration
+ *** about to expire - certificate is still valid. If certificate lifespan is week or smaller, this rougly corresponds to day to EOL, otherwise 7 days to EOL
+ *** valid - more than above boundries
+ * add config options to handle:
+ ** frequency of warning
+ *** 0 - turn off - one time check on startup as in previous version
+ *** n+ - millisecconds between periodic checks
+
+
+=== Changed requirements
+
+N/A
+
+
+=== Non-Requirements
+
+N/A
+
+=== Future Work
+
+N/A
+
+== Backwards Compatibility
+
+No. This feature should not introduce incompatibility, since its just logging, which is simple config adjustment.
+
+NOTE: above is subject to change based on feedback( default to 0, turn-off periodic so its essentially like this change wasnt introduced)
+
+=== Default Configuration
+
+__<Does the proposed work change the default value of any current configuration attributes? Does it change the configuration generated by any current Galleon layers?>__
+
+=== Importing Existing Configuration
+
+Should default to predefined values.
+
+=== Deployments
+
+No impact.
+
+=== Interoperability
+
+__<Is this feature impacting interoperability?>__
+No.
+
+== Implementation Plan
+
+N/A
+
+== Admin Clients
+
+CLI should not be affected. However, HAL will most likely require follow up, since this feature will introduce new model params/context. Given defaults, its not a breaking change.
+
+== Security Considerations
+
+Slightly increase security risk compared to current state. If attacker has access to log/events it will provide view of current health of server, rather than possible peek on startup. Convenient, but not something that could not be circumvented without this feature.
+
+[[test_plan]]
+== Test Plan
+
+Integration tests should be fairly acceptable.
```

Comment:
We should expand on this section, we have a number of supported KeyStore types and some of these even have sub types so we should specify the tests and the permutations we will run.